local-process-quickstart.sh

```diff
@@ -0,0 +1,160 @@
+   fi
+   exit 1
+}
+
+while getopts "g:n:r:c" opt
+do
+   case "$opt" in
+      c ) CLEANUP="true"  ;;
+      g ) RGNAME="$OPTARG"  ;;
+      n ) AKSNAME="$OPTARG"  ;;
+      r ) REPOROOT="$OPTARG"  ;;
+      ? ) helpFunction ;; # Print helpFunction in case parameter is non-existent
+   esac
+done
+
+# Print helpFunction in case parameters are empty
+if [ ! -z "$CLEANUP" ]
+then
+   if [ -z "$RGNAME" ]
+   then
+      echo "Please pass -g when calling -c"
+      helpFunction
+   else
+      cleanupFunction
+   fi
+elif [ -z "$RGNAME" ] || [ -z "$AKSNAME" ]
+then
+   echo "Some or all of the parameters are empty";
+   helpFunction
+fi
+
+if [ -z "$REPOROOT" ]
+then
+   echo "Defaulting Dev spaces repository root to current directory : ${PWD}"
+   REPOROOT=$PWD
+fi
+
+installHelmFunction
+ 
+echo "Setting the Kube context"
+az aks get-credentials -g $RGNAME -n $AKSNAME
+ 
+PUBLICIP=$(az network public-ip create --resource-group $RGNAME --name $PIPNAME --sku Standard --allocation-method static --query publicIp.ipAddress -o tsv)
+echo "BikeSharing ingress Public ip: " $PUBLICIP
+ 
+echo "Create namespace ${INGRESSNAME}"
+kubectl create namespace $INGRESSNAME
+ 
+# Use Helm to deploy a traefik ingress controller
+echo "helm repo add && helm repo update"
+${HELMDIR}/helm repo add stable https://kubernetes-charts.storage.googleapis.com/
+${HELMDIR}/helm repo update
+echo "helm install traefik ingress controller"
+kubectl api-versions | grep "rbac.authorization.k8s.io"
+if [[ $? -eq 0 ]]
+then
+   SUB=$(az account show --query id | cut -d'"' -f2)
+   SPID=$(az aks show -n ${AKSNAME} -g ${RGNAME} --query servicePrincipalProfile.clientId | cut -d'"' -f2)
+   az role assignment create --assignee ${SPID} --scope /subscriptions/${SUB}/resourceGroups/${RGNAME} --role "Network Contributor"
+   ${HELMDIR}/helm install $INGRESSNAME stable/traefik \
+      --namespace $INGRESSNAME \
+      --set kubernetes.ingressClass=traefik \
+      --set fullnameOverride=$INGRESSNAME \
+      --set rbac.enabled=true \
+      --set service.annotations."service\.beta\.kubernetes\.io/azure-load-balancer-resource-group"="$RGNAME" \
+      --set loadBalancerIP=$PUBLICIP \
+      --set kubernetes.ingressEndpoint.useDefaultPublishedService=true \
+      --version 1.85.0
+else
+   ${HELMDIR}/helm install $INGRESSNAME stable/traefik \
+      --namespace $INGRESSNAME \
+      --set kubernetes.ingressClass=traefik \
+      --set fullnameOverride=$INGRESSNAME \
+      --set rbac.enabled=false \
+      --set service.annotations."service\.beta\.kubernetes\.io/azure-load-balancer-resource-group"="$RGNAME" \
+      --set loadBalancerIP=$PUBLICIP \
+      --set kubernetes.ingressEndpoint.useDefaultPublishedService=true \
+      --version 1.85.0
+fi
+ 
+NIPIOFQDN=${PUBLICIP}.nip.io
+echo "The Nip.IO FQDN would be " $NIPIOFQDN
+ 
+CHARTDIR=${REPOROOT}/samples/BikeSharingApp/charts/
+echo "---"
+echo "Chart directory: $CHARTDIR"
+ 
+echo "Create namespace dev"
+kubectl create ns dev
```

Comment:
What if this one already exists? "dev" is quite a common name. I'm specifically concerned by the clean up phase where we could remove as well user's pre-existing services.
Please consider adding a check that warns the user if the namespace "dev" already exists.

Reply:
That's a good point... I don't think we should delete the dev namespace in that case...

Reply:
Added a confirmation message asking the user if they want to delete the ns